org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/TypeBinding.java

```diff
@@ -588,10 +588,16 @@ public int getKind() {
 	public int getModifiers() {
 		if (isClass()) {
 			ReferenceBinding referenceBinding = (ReferenceBinding) this.binding;
-			final int accessFlags = referenceBinding.getAccessFlags() & VALID_MODIFIERS;
+			int k = referenceBinding.getAccessFlags();
+			final int accessFlags = k & VALID_MODIFIERS;
 			if (referenceBinding.isAnonymousType()) {
 				return accessFlags & ~Modifier.FINAL;
+			} else if (referenceBinding.isSealed()) {
+				return Modifier.SEALED;
```

Comment:
This is wrong. With this, we are discarding every other bit in accessFlags.

Reply:
Done